cmd/scheduler.go

```diff
@@ -100,45 +185,58 @@ func addCommands() {
-	// so we simply prevent that from ever happening here by placing them in the same pool.
-	import_update_pool := "import_update"
-
-	// Jobs that run once a day (at night), and drain a machine's resources
-	heavy_daily_pool := "heavy_daily"
+	// Convert time from string into time.Duration format
+	interval, err := time.ParseDuration(*cmd.Interval)
+	if err != nil {
+		logger.Errorf("Error parsing interval for task '%s': %v", cmd.Name, err)
+	}
+	timeout, err := time.ParseDuration(*cmd.Timeout)
+	if err != nil {
+		logger.Errorf("Error parsing timeout for task '%s': %v", cmd.Name, err)
+	}
 
-	// Imports
-	add(true, "Locator", import_update_pool, 15, 2*hour, "c:\\imqsbin\\bin\\imqstool", "locator", "imqs", "!LOCATOR_SRC", "c:\\imqsvar\\staging", "!JOB_SERVICE_URL", "!LEGACY_LOCK_DIR")
-	add(true, "ImqsTool Importer", import_update_pool, 15, 6*hour, "c:\\imqsbin\\bin\\imqstool", "importer", "!LEGACY_LOCK_DIR", "!JOB_SERVICE_URL")
-	add(true, "Docs Importer", import_update_pool, 15, 2*hour, "ruby", "c:\\imqsbin\\jsw\\ImqsDocs\\importer\\importer.rb")
+	newjob := add(isEnabled, *cmd.Name, *cmd.Pool, interval, timeout, *cmd.Command, *cmd.Params)
+	if cmd.Start_Time != nil {
+		start_time, err := time.ParseDuration(*cmd.Start_Time)
+		if err != nil {
+			logger.Errorf("Error parsing start_time for task '%s': %v", cmd.Name, err)
+		}
+		hours := int(start_time / time.Hour)
+		start_time -= time.Duration(hours) * time.Hour
+		minutes := int(start_time / time.Minute)
+		newjob.SetStartTime(hours, minutes)
+	}
+}
 
-	// Updaters
-	add(true, "ImqsConf Update", import_update_pool, 5*minute, 30*minute, "c:\\imqsbin\\cronjobs\\update_runner.bat", "conf -prod")
-	add(true, "ImqsBin Update", import_update_pool, 5*minute, 2*hour, "c:\\imqsbin\\cronjobs\\update_runner.bat", "imqsbin -prod")
+func loadCommandsFromStaticConfig() {
+	if err := staticConfig.LoadFile("c:/imqsbin/static-conf/scheduled-tasks.json"); err != nil {
+		logger.Errorf("Error loading static config file 'scheduled-tasks.json': %v", err)
+	}
 
-	// Other
-	add(true, "Ping", "ping", minute, 5*minute, "ruby", "c:\\imqsbin\\cronjobs\\ping_services.rb")
-	add(true, "Auth Log Scraper", "logscrape", 24*hour, 24*hour, "ruby", "c:\\imqsbin\\cronjobs\\logscrape.rb")
-	add(false, "Theme saver", "theme_saver", minute, 5*minute, "ruby", "c:\\imqsbin\\conftools\\theme_saver.rb", "prepare_for_edit")
+	// Build map of enabled jobs
+	var enabledMap map[string]bool = make(map[string]bool)
+	for _, e := range staticConfig.Enabled {
+		enabledMap[e] = true
+	}
+	for _, e := range staticConfig.Disabled {
+		enabledMap[e] = false
+	}
 
-	// Heavy daily pool
-	backup := add(true, "Backup", heavy_daily_pool, daily, 12*hour, "ruby", "c:\\imqsbin\\cronjobs\\backup_v8.rb")
-	backup.SetStartTime(23, 0)
+	// Rebuild array of commands using values from config file
+	commands = nil
+	for _, t := range staticConfig.Tasks {
+		isEnabled := false
```

Comment:
Something is not quite right here: I cannot see how name should be allowed to be nil. If it is the config is invalid.  I suggest that you fail if name is nil and then simply call addCommand(t, enabledMap[*t.Name])??